app/ports/migrations/0004_add-fields-to-stats.py

```diff
@@ -0,0 +1,41 @@
+# Generated by Django 2.1.7 on 2019-07-27 13:03
+
+import django.contrib.postgres.fields.jsonb
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('ports', '0003_add-field-cxx_stdlib'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='submission',
+            name='build_arch',
+            field=models.CharField(default='', max_length=20),
```

Comment:
Can the migration file contain something like the following (pseudocode):
* for all existing entries where os_version in (10.7, 10.8, 10.9, ... 10.14) set build_arch = x86_64
* for all existing entries where os_arch = ppc, set build_arch = ppc
* for all existing entries where os_version = 10.5 and os_arch = i386, set build_arch = i386

Reply:
Do we need to supply such code with the repo? I was planning to do this, but not commit the code.

Reply:
I'm not sure. In my understanding one of the ideas of migrations is to do precisely that: add missing tables, columns, and populate them with some values. But then again, since we have only been collecting statistics at a single site, there will be precisely one user of that migration anyway, so it boils down how you want to do it.

(The same is true for all migrations mostly, and the same argument applies that we hardly need migrations during the development period.)

In principle I'm also fine with manually running the changes. You had 160 submissions so far. I would personally probably not be manually editing each one of them, and if you run slightly more complex queries, it doesn't make much of a difference whether you just execute them, or store them first, then run migrations to auto-populate everything.

I leave it up to you to decide.

@umeshksingla?

Reply:
I'd just run the django migrations even if it's adding the default value to all rows. Because we have to add it to the model anyway and eventually migration files so it'll try to overwrite them.